File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/addit/payload/PayloadAdapter.java
package com.adadapted.android.sdk.core.addit.payload;

import com.adadapted.android.sdk.core.addit.Content;

import org.json.JSONObject;

import java.util.List;

public interface PayloadAdapter {
    interface Callback {
        void onSuccess(List<Content> content);
        void onFailure(String message);
    }

    void pickup(JSONObject request, Callback callback);
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/payload/PayloadAdapter.kt
package com.adadapted.android.sdk.core.payload

import com.adadapted.android.sdk.core.atl.AdditContent
import com.adadapted.android.sdk.core.device.DeviceInfo

interface PayloadAdapter {
    suspend fun pickup(deviceInfo: DeviceInfo, callback: (content: List<AdditContent>) -> Unit)
    suspend fun publishEvent(deviceInfo: DeviceInfo, event: PayloadEvent)
}